HW/feature_eng.py

```diff
@@ -0,0 +1,61 @@
+import sys
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+# user input to receive dataset
+# https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data
+dataset = pd.read_csv(
+    "https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data"
+)
+# specifing the response variable
+response_name = " <=50K"
+# Don't include response in other variables, we don't want to plot response vs response.
+col_name = [col for col in dataset.columns if col != response_name]
+
+
+def predictor_response_plots(col_name):
+
+    if (dataset[response_name].dtypes == "object") & (
+        dataset[col_name].dtypes == "object"
+    ):
+        # Heatmap ( categorical variables vs categorical response)
+        cat_freq = pd.crosstab(index=dataset[col_name], columns=dataset[response_name])
+        sns.heatmap(data=cat_freq, annot=True, fmt=".1f")
+        plt.show()
```

Comment:
If you use plotly here you can do the following to save it as an html file pretty easy.

from plotly.io import to_html
fig = ### set up plot here
html = fig.to_html()
with open('html-filename-here.html') as outfile:
    outfile.write(html)